scipy/linalg/basic.py

```diff
@@ -48,33 +120,19 @@ def solve(a, b, sym_pos=False, lower=False, overwrite_a=False,
     corresponding string to ``assume_a`` key chooses the dedicated solver.
     The available options are
 
-    ===================  ========
-     generic matrix       'gen'
-     symmetric            'sym'
-     hermitian            'her'
-     positive definite    'pos'
-    ===================  ========
+    <================ FILL IN ALL THE OPTIONS ==================>
 
-    If omitted, ``'gen'`` is the default structure.
-
-    The datatype of the arrays define which solver is called regardless
-    of the values. In other words, even when the complex array entries have
-    precisely zero imaginary parts, the complex solver will be called based
-    on the data type of the array.
 
     Parameters
     ----------
     a : (N, N) array_like
         Square input data
     b : (N, NRHS) array_like
         Input data for the right hand side.
-    sym_pos : bool, optional
-        Assume `a` is symmetric and positive definite. This key is deprecated
-        and assume_a = 'pos' keyword is recommended instead. The functionality
-        is the same. It will be removed in the future.
-    lower : bool, optional
-        If True, only the data contained in the lower triangle of `a`. Default
-        is to use upper triangle. (ignored for ``'gen'``)
+    assume_a : str, optional
```

Comment:
In the current version of the code, this isn't used any more, right? Do you plan on bringing it back and skipping to the appropriate solver if it's supplied? (Maybe the default would be `None`?)

Reply:
it will be used to bypass the structure checks if you already know what you have but currently I don't have names or abbreviations for them so it's left for later when the code stabilizes.